needle-notion-connector/src/server/backend/connectors-backend.ts

```diff
@@ -0,0 +1,154 @@
+import type {
+  ConnectorRequest,
+  CreateConnectorRequest,
+} from "~/models/connectors-models";
+import {
+  createConnector,
+  getConnector,
+  listConnectors,
+  deleteConnector,
+  publishConnectorRun,
+  createNeedleFileId,
+  type Session,
+  type ConnectorRunDescriptor,
+} from "@needle-ai/needle-sdk";
+import { db } from "../db";
+import { eq } from "drizzle-orm";
+import { notionConnectorsTable, notionPagesTable } from "../db/schema";
+import { Client as NotionClient } from "@notionhq/client";
+import type {
+  DatabaseObjectResponse,
+  PageObjectResponse,
+} from "@notionhq/client/build/src/api-endpoints";
+
+export async function createNotionConnector(
+  request: CreateConnectorRequest,
+  session: Session,
+) {
+  const connector = await createConnector(
+    {
+      name: request.name,
+      cronJob: request.cronJob,
+      cronJobTimezone: request.cronJobTimezone,
+      collectionIds: [request.collectionId],
+      credentials: request.notionToken.access_token,
+    },
+    session.id,
+  );
+
+  const pagesToInsert = request.notionPages.map((p) => ({
+    ndlConnectorId: connector.id,
+    ndlFileId: createNeedleFileId(),
+    notionUrl: p.url,
+    notionPageId: p.id,
+    notionLastEditedTime: p.last_edited_time,
+  }));
+  await db.insert(notionPagesTable).values(pagesToInsert);
+
+  const connectorsToInsert = request.notionPages.map((p) => ({
+    ndlConnectorId: connector.id,
+    notionWorkspaceId: request.notionToken.workspace_id,
+    notionWorkspaceName: request.notionToken.workspace_name,
+    notionUserId: request.notionToken.owner.user.id,
+    notionPageId: p.id,
+  }));
+  await db.insert(notionConnectorsTable).values(connectorsToInsert);
+
+  await runNotionConnector({ connectorId: connector.id }, session);
+
+  return connector;
+}
+
+export async function listNotionConnectors(session: Session) {
+  return await listConnectors(session.id);
+}
+
+export async function getNotionConnector(
+  { connectorId }: ConnectorRequest,
+  session: Session,
+) {
+  const connector = await getConnector(connectorId, session.id);
+
+  const files = await db
+    .select()
+    .from(notionPagesTable)
+    .where(eq(notionPagesTable.ndlConnectorId, connectorId));
+
+  return { ...connector, files };
```

Comment:
as discussed in the call - here we'd like to add the config of the connector, a.k.a. workspaces we are tracking.